samples/azure/azure-service-fabric-routing/sample.md

```diff
@@ -0,0 +1,202 @@
+
+snippet: ApplyPartitionConfigurationToEndpoint-ZipCodeVoteCount
+
+### Local sends
+
+All local sends are handled by the `PartitionAwareDistributionStrategy`. For each locally sent message it sets the `partition-key` header value to the local partition key and routes the message to the queue associated with the local partition.
+
+### Replies
+
+When replying, an endpoint routes the reply message to the endpoint that initiated the conversation. It's the responsibility of the requestor to properly set the reply-to header before sending out the request. For a partitioned endpoint this implies that it sets the reply to header to its instance specific queue instead of the shared queue. This functionality is covered by the `HardcodeReplyToAddressToLogicalAddress` behavior.
+
+
+## Receiver Side Distribution
+
+Receiver Side Distribution validates if a replica of a partitioned endpoint can process an incoming message or should forward it to the appropriate replica instead. Partition validation is performed by inspecting message headers and the message body.
+
+A partitioned endpoint can be configured to check that an incoming message should be processed locally. If it is not the case, the message is forwarded to the correct partition.
+
+### Message headers inspection
+
+Every incoming message has its `partition-key` header value inspected by the `DistributeMessagesBasedOnHeader` behavior. If the value specified in the header is matching the receiver`s partition key, then the message is processed. Otherwise, the message is forwarded to the appropriate partition specified by the header value. If the partition key value indicates non-existent partition, the message is moved to the error queue.
+
+If the `partition-key` header does not exist, the pipeline execution continues to the *Message body inspection* step.
+
+### Message body inspection
+
+If message headers inspection is unsuccessful with value extraction of `partition-key` header, then the message body is used to determine the partition key value. The `DistributeMessagesBasedOnPayload` behavior determines the partition value using the mapping function provided via the configuration API. The result of the mapping function is stored as the `partition-key` header value. The inspection logic can raise a `PartitionMappingFailedException` exception if the partition key is not a valid partition key for a given endpoint. For a partition key that is not local, the message is forwarded to the appropriate partition. 
+
+The forwarding/processing decision is made in the same way as in the *Header inspection* step.
+
+NOTE: `PartitionMappingFailedException` is considered an [unrecoverable exception](/nservicebus/recoverability/custom-recoverability-policy.md). Whenever such an exception is raised the message that triggered the exception will be moved immediately to the configured error queue.
```

Comment:
Can be a bit shorter I think: 

The `PartitionMappingFailedException` is considered an [unrecoverable exception](/nservicebus/recoverability/custom-recoverability-policy.md) and the message will be moved to the error queue immediately.